compress/compress_test.go

```diff
@@ -407,58 +417,72 @@ func benchmarkCompression(b *testing.B, codec pkg.Codec, buf *bytes.Buffer, payl
 	return 1 - (float64(buf.Len()) / float64(len(payload)))
 }
 
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
 func makeTopic() string {
 	return fmt.Sprintf("kafka-go-%016x", rand.Int63())
 }
 
-func createTopic(t *testing.T, partitions int) string {
+func newLocalClientAndTopic() (*kafka.Client, string, func()) {
```

Comment:
seems to mostly be a copy of what's in client_test.go, would it help to move this to the `testing` package too?

Reply:
Unfortunately it would result in a cycle between the top level `kafka` package and `kafka/testing`, we'd have to refactor the tests to all be in the `kafka_test` package for that to work.